### ISSUE TITLE
Only render source that is not folded

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/atom-asciidoc-preview#40]
This might be a shoot for the moon sort of request, but it would be cool if we could figure out how to fold (hide) the source on blocks (like sections or block content) so that you only see content in the open regions.

There's some sort of [fold](https://atom.io/docs/api/v0.130.0/Selection#instance-fold) method in Atom. Maybe that can make it possible.

**Comment by nicorikken** [comment]
Does this relate to the AsciiDoc source, or the rendered source blocks? The issue regarding the AsciiDoc source folding is also an outstanding issue in the atom-language-asciidoc: project https://github.com/asciidoctor/atom-language-asciidoc/issues/7

**Comment by mojavelinux** [comment]
Absolutely. This is definitely something that fits with the language package, not the preview. I think we can close this issue here.

Once the folding works in the editor part, we may want to also hide parts from the preview that are folded. But we'll address that idea separately.

**Comment by mojavelinux** [comment]
Actually, instead of closing this issue, I'm going to update the title to reflect a new purpose.

**Comment by nicorikken** [comment]
:+1:

**Comment by mojavelinux** [comment]
This issue depends on source code folding being implemented in the language package.

**Comment by nicorikken** [comment]
And some form a state synchronization is desired, or replacing the code block shown with a folding placeholder (like '...') requiring reloading of the preview.
In that sense it can relate to place synchronization, for which there is a dedicated Markdown helper, to sync place between source code and preview.

**Comment by mojavelinux** [comment]
Using a placeholder is one idea. Another idea is to simply feed the preview the visible source so that what you see rendered is only a partial document. I imagine it should be possible for the preview to query for / calculate the source that is not folded. Maybe.

**Comment by mojavelinux** [comment]
I'm going to mark this one as "future" since there are many other valuable improvements to address first, and because we can't act on it until we get folds working in the source.
